src/app/types/url.ts

```diff
@@ -18,14 +21,9 @@ export function getIDFromURL(song: Song): string {
 	}
 }
 
-export function getEmbedURL(song: Song, sanitizer: DomSanitizer): SafeResourceUrl {
-	switch (song.platform) {
-		case SongPlatform.YOUTUBE: {
-			const videoID = getIDFromURL(song);
-			return sanitizer.bypassSecurityTrustResourceUrl(`https://youtube.com/embed/${videoID}`);
-		}
-
-		default:
-			return "";
-	}
+export function URLToSongPlatform(url: string): SongPlatform {
+	const hostname = new URLParse(url, true).hostname;
+	if (YOUTUBE_URLs.includes(hostname)) return SongPlatform.YOUTUBE;
```

Comment:
je suis pas sur mais est-ce-que éventuellement on pourrais faire un dict du genre
```
{
  name : "youtube"
  enum: SongPlatform.Youtube
  URLs: YOUTUBE_URLS
}
```
pour qu'on puisse par la suite itérer dessus? on fera juste un `if iter.URLS.include(hostname) return iter.enum`

Reply:
Comme discuté, ça ne semble pas nécessaire dans le cas où on a juste YOUTUBE ou SOUNDCLOUD. On pourra revoir ça si on souhaite ajouter d'autres fournisseurs.